src/main/java/com/conveyal/datatools/manager/controllers/api/LabelController.java

```diff
@@ -0,0 +1,218 @@
+     *
+     * FIXME: Should this just set a "deleted" flag instead of removing from the database entirely?
+     */
+    private static Label deleteLabel(Request req, Response res) {
+        Label label = requestLabelById(req, Actions.MANAGE);
+
+        // More specific error message if label doesn't exist
+        if (label == null) {
+            logMessageAndHalt(req, 403, "Label does not exist.");
+            return null;
+        }
+
+        try {
+            // Iterate over feed sources and remove any references to the label
+            Persistence.feedSources.getAll().forEach(feedSource -> {
+                List<String> newLabels = feedSource.labels.stream()
+                        .filter(l -> !l.equals(label.id))
+                        .collect(Collectors.toList());
+                // If there are changes, update the feed source
+                if (!newLabels.equals(feedSource.labels)) {
+                    FeedSource fs = Persistence.feedSources.getById(feedSource.id);
+                    fs.labels = newLabels;
+                    Persistence.feedSources.replace(fs.id, fs);
+                }
+            });
+
+            label.delete();
+            return label;
+        } catch (Exception e) {
+            logMessageAndHalt(req, 500, "Unknown error occurred while deleting label.", e);
+            return null;
+        }
+    }
+
+    /**
+     * Helper function returns label if user has permission for specified action.
+     * @param req spark Request object from API request
+     * @param action action type (either "view" or Permission.MANAGE)
+     * @return label object for ID
+     */
+    private static Label requestLabelById(Request req, Actions action) {
+        String id = req.params("id");
+        if (id == null) {
+            logMessageAndHalt(req, 400, "Please specify id param");
+        }
+        Label label = Persistence.labels.getById(id);
+        checkLabelPermissions(req, label, action);
+
+        return label;
+    }
+
+    /** Helper method will halt execution and throw an error message if a given
+     * label is not allowed to be accessed by the current user (taken from the req object)
+     *
+     * @param req       Spark request used for determining user permissions
+     * @param label     Label to check
+     * @param action    Action to be taken on label, which changes who can do what
+     */
+    public static void checkLabelPermissions(Request req, Label label, Actions action) {
+        Auth0UserProfile userProfile = req.attribute("user");
+        // check for null label
+        if (label == null) {
+            logMessageAndHalt(req, 400, "Label ID does not exist");
+            return;
+        }
+
+        boolean isProjectAdmin = userProfile.canAdministerProject(label.projectId, label.organizationId());
+        switch (action) {
+            case CREATE:
+            case MANAGE:
+            case EDIT:
+                // Only project admins can edit labels
```

Comment:
Should this say:

// Only project admins can create, manage and edit labels

Reply:
Good catch thank you!